flash/data/base_viz.py

```diff
@@ -1,101 +1,139 @@
 
-    def on_pre_tensor_transform(self, sample: Any, running_stage: RunningStage) -> None:
-        self._store(sample, "pre_tensor_transform", running_stage)
+            @staticmethod
+            def configure_data_fetcher(*args, **kwargs) -> BaseDataFetcher:
+                return CustomBaseViz(*args, **kwargs)
 
-    def on_to_tensor_transform(self, sample: Any, running_stage: RunningStage) -> None:
-        self._store(sample, "to_tensor_transform", running_stage)
+        dm = CustomImageClassificationData.from_folders(
+            train_folder="./data/train",
+            val_folder="./data/val",
+            test_folder="./data/test",
+            predict_folder="./data/predict")
 
-    def on_post_tensor_transform(self, sample: Tensor, running_stage: RunningStage) -> None:
-        self._store(sample, "post_tensor_transform", running_stage)
+        # visualize a ``train`` batch
+        dm.show_train_batches()
 
-    def on_per_batch_transform(self, batch: Any, running_stage: RunningStage) -> None:
-        self._store(batch, "per_batch_transform", running_stage)
+        # visualize next ``train`` batch
+        dm.show_train_batches()
 
-    def on_collate(self, batch: Sequence, running_stage: RunningStage) -> None:
-        self._store(batch, "collate", running_stage)
+        # visualize a ``val`` batch
+        dm.show_val_batches()
 
-    def on_per_sample_transform_on_device(self, samples: Sequence, running_stage: RunningStage) -> None:
-        self._store(samples, "per_sample_transform_on_device", running_stage)
+        # visualize a ``test`` batch
+        dm.show_test_batches()
 
-    def on_per_batch_transform_on_device(self, batch: Any, running_stage: RunningStage) -> None:
-        self._store(batch, "per_batch_transform_on_device", running_stage)
+        # visualize a ``predict`` batch
+        dm.show_predict_batches()
 
-    @contextmanager
-    def enable(self):
-        self.enabled = True
-        yield
-        self.enabled = False
+    .. note::
 
-    def attach_to_datamodule(self, datamodule) -> None:
-        datamodule.viz = self
+        If the user wants to plot all different transformation stages at once,
+        override the ``show`` function directly.
+
+    Example::
+
+        class CustomBaseViz(BaseViz):
+
+            def show(self, batch: Dict[str, Any], running_stage: RunningStage):
+                print(batch)
+                # out
+                {
+                    'load_sample': [...],
+                    'pre_tensor_transform': [...],
+                    'to_tensor_transform': [...],
+                    'post_tensor_transform': [...],
+                    'collate': [...],
+                    'per_batch_transform': [...],
+                }
+
+    .. note::
+
+        As the :class:`~flash.data.process.Preprocess` hooks are injected within
+        the threaded workers for the DataLoader,
+        the data won't be accessible when using ``num_workers > 0``.
```

Comment:
can this somewhere be detected, to raise a warning or even an error ?

Reply:
Not sure what you mean. There is no issue. Only for visualization, but I have a fix coming for it